PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/utility/AzureAnAction.java

```diff
@@ -104,15 +104,19 @@ protected String getOperationName(AnActionEvent event) {
      * @return
      */
     private String transformHDInsight(String serviceName, AnActionEvent event) {
-        if (serviceName.equals(TelemetryConstants.ACTION)) {
-            String text = event.getPresentation().getText().toLowerCase();
-            if (text.contains("spark") || text.contains("hdinsight")) {
-                return TelemetryConstants.HDINSIGHT;
-            }
-            String place = event.getPlace().toLowerCase();
-            if (place.contains("spark") || place.contains("hdinsight")) {
-                return TelemetryConstants.HDINSIGHT;
+        try {
+            if (serviceName.equals(TelemetryConstants.ACTION)) {
+                String text = event.getPresentation().getText().toLowerCase();
+                if (text.contains("spark") || text.contains("hdinsight")) {
+                    return TelemetryConstants.HDINSIGHT;
+                }
+                String place = event.getPlace().toLowerCase();
+                if (place.contains("spark") || place.contains("hdinsight")) {
+                    return TelemetryConstants.HDINSIGHT;
+                }
             }
+            return serviceName;
+        } catch (Exception ignore) {
```

Comment:
Should we do NPE verification or catch all exceptions?

Reply:
I think we should just swallow all exception. Here is just for metrics, should not block our main process